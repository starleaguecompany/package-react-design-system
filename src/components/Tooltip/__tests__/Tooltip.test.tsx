import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'

import { Tooltip } from '../../../index'
import userEvent from '@testing-library/user-event'

describe('Tooltip', () => {
  test('should render correctly with no props', async () => {
    render(
      <Tooltip>
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    expect(screen.getByText('Test tooltip')).toHaveAttribute('tabindex', '0')
  })

  test('should render correctly with attributes', async () => {
    render(
      <Tooltip id="test-id" className="test-class">
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    expect(screen.getByTestId('Tooltip')).toHaveClass('test-class')
    expect(screen.getByTestId('Tooltip')).toHaveAttribute('id', 'test-id')
    expect(screen.getByText('Test tooltip')).toBeVisible()
  })

  test('should render correctly with title and visible on hover', async () => {
    render(
      <Tooltip title="Tooltip title">
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    userEvent.unhover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.queryByText('Tooltip title')).not.toBeInTheDocument())
    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByText('Tooltip title')).toBeVisible())
  })

  test('should render correctly with content and visible on hover', async () => {
    render(
      <Tooltip content="Tooltip content">
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    userEvent.unhover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.queryByText('Tooltip content')).not.toBeInTheDocument())
    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByText('Tooltip content')).toBeVisible())
  })

  test('should render correctly with footer and visible on hover', async () => {
    render(
      <Tooltip footer="Tooltip footer">
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    userEvent.unhover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.queryByText('Tooltip footer')).not.toBeInTheDocument())
    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByText('Tooltip footer')).toBeVisible())
  })

  test('should render correctly with width and visible on hover', async () => {
    render(
      <Tooltip width={400}>
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    expect(screen.getByTestId('Tooltip')).toHaveStyle('width: 400px')
  })

  test('should render correctly with placement prop', async () => {
    render(
      <Tooltip title="Tooltip title" placement="right">
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    const bottomTooltip = document.querySelector('[data-popper-placement="right"]')
    expect(bottomTooltip).toBeInTheDocument()
  })

  test('should hide tooltip on end of hover', async () => {
    render(
      <Tooltip title="Title">
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    userEvent.unhover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.queryByText('Title')).not.toBeInTheDocument())
  })

  test('should render correctly with closable prop', async () => {
    render(
      <Tooltip title="Title" closable>
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    expect(screen.getByTestId('Tooltip')).toContainElement(screen.getByTestId('Cross'))
  })

  test('should execute callback on clicking cross icon', async () => {
    const handleClose = jest.fn()

    render(
      <Tooltip title="Title" onClose={handleClose} closable>
        <span>Test tooltip</span>
      </Tooltip>
    )

    userEvent.hover(screen.getByText('Test tooltip'))
    await waitFor(() => expect(screen.getByTestId('Tooltip')).toMatchSnapshot())
    userEvent.unhover(screen.getByText('Test tooltip'))
    expect(handleClose).toBeCalledTimes(0)
    userEvent.hover(screen.getByText('Test tooltip'))
    userEvent.click(screen.getByTestId('Cross'))
    expect(handleClose).toBeCalledTimes(1)
  })
})
