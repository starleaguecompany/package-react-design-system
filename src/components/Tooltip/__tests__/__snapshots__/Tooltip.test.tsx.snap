// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should execute callback on clicking cross icon 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  >
    <div
      class="container direction-horizontal justify-space-between"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="size-16 strong"
      >
        Title
      </div>
      <span
        class="close container size-16 shape-round"
        data-qa="Icon"
      >
        <svg
          data-qa="Cross"
          fill="none"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0)"
          >
            <path
              d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
              stroke="currentColor"
              stroke-dasharray="10 5"
            />
          </g>
          <defs>
            <clippath
              id="clip0"
            >
              <path
                d="M0 0h20v20H0z"
                fill="currentColor"
              />
            </clippath>
          </defs>
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`Tooltip should hide tooltip on end of hover 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  >
    <div
      class="container direction-horizontal justify-space-between"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="size-16 strong"
      >
        Title
      </div>
    </div>
  </div>
</div>
`;

exports[`Tooltip should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="Tooltip"
  id="test-id"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  />
</div>
`;

exports[`Tooltip should render correctly with closable prop 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  >
    <div
      class="container direction-horizontal justify-space-between"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="size-16 strong"
      >
        Title
      </div>
      <span
        class="close container size-16 shape-round"
        data-qa="Icon"
      >
        <svg
          data-qa="Cross"
          fill="none"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0)"
          >
            <path
              d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
              stroke="currentColor"
              stroke-dasharray="10 5"
            />
          </g>
          <defs>
            <clippath
              id="clip0"
            >
              <path
                d="M0 0h20v20H0z"
                fill="currentColor"
              />
            </clippath>
          </defs>
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`Tooltip should render correctly with content and visible on hover 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  >
    <div
      class="size-14"
    >
      Tooltip content
    </div>
  </div>
</div>
`;

exports[`Tooltip should render correctly with footer and visible on hover 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  >
    Tooltip footer
  </div>
</div>
`;

exports[`Tooltip should render correctly with no props 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  />
</div>
`;

exports[`Tooltip should render correctly with placement prop 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  >
    <div
      class="container direction-horizontal justify-space-between"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="size-16 strong"
      >
        Tooltip title
      </div>
    </div>
  </div>
</div>
`;

exports[`Tooltip should render correctly with title and visible on hover 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 300px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  >
    <div
      class="container direction-horizontal justify-space-between"
      data-qa="Space"
      style="column-gap: 8px; row-gap: 8px;"
    >
      <div
        class="size-16 strong"
      >
        Tooltip title
      </div>
    </div>
  </div>
</div>
`;

exports[`Tooltip should render correctly with width and visible on hover 1`] = `
<div
  class="container"
  data-qa="Tooltip"
  style="width: 400px;"
>
  <div
    class="container direction-vertical"
    data-qa="Space"
    style="column-gap: 8px; row-gap: 8px;"
  />
</div>
`;
